src/components/TodoList.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from "react";
+import uuid from "uuid";
+import Todo from "./Todo";
+import TodoInput from "./TodoInput";
+import SortTodosForm from "./SortTodosForm";
+import FilterTodosSection from "./FilterTodosSection";
+import DeleteTodosSection from "./DeleteTodosSection";
+
+export default class TodoList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      todos: [
+        { task: "have some coffee", id: uuid(), completed: false },
+        { task: "walk the dog", id: uuid(), completed: false },
+        { task: "take over the world", id: uuid(), completed: false }
+      ],
+      displayTodos: "all",
+      sortByStatus: false
+    };
+  }
+
+  createTodo = newTodo => {
+    this.setState({
+      todos: [...this.state.todos, newTodo]
```

Comment:
tak nie rób, Nie orzekazuj do setState nic co bezpośrednio bazuje na this.state. 
W większości przypadków jest to groźne, ale lepiej od razu robić poprawnie

https://medium.com/@wereHamster/beware-react-setstate-is-asynchronous-ce87ef1a9cf3

setState nie zmienia this.state od razu. Więc this.state może być nieaktualny.

W takiej sytuacji, jeśli w setState musisz bazować na poprzednim stanie trzeba to zrobić przez callback.

```
this.setState((prevState) => {
   return {
       todos: [...prevState.todos, newTodo]

     }

})